pkg/ingester/client/timeseries.go

```diff
@@ -70,27 +70,36 @@ type LabelAdapter labels.Label
 
 // Marshal implements proto.Marshaller.
 func (bs *LabelAdapter) Marshal() ([]byte, error) {
-	buf := make([]byte, bs.Size())
-	_, err := bs.MarshalTo(buf)
-	return buf, err
+	size := bs.Size()
+	buf := make([]byte, size)
+	n, err := bs.MarshalToSizedBuffer(buf[:size])
+	return buf[:n], err
```

Comment:
Not sure this is correct. Looking at others generated `*.pb.go` files, I believe it should be:

```
if err != nil {
    return nil, err
}
return buf[:n], nil
``` 

Reply:
yes good catch, although not sure it matter, but will update.